### PR TITLE
fix(theming): fix read-only $mdThemingProvider.configuration()

### PR DESCRIPTION
**Pull request by ThomasBurleson**
*  insure we clone properly the a configuration map object.
*  implement unit test to verify.

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
@jelbourn - ping

File: src/core/services/theming/theming.js [hunk 1]
```diff
@@ -331,7 +331,11 @@ function ThemingProvider($mdColorPalette, $$mdMetaProvider) {
     configuration : function() {
       // return a read-only clone of the current configuration
       var locals = { defaultTheme : defaultTheme, alwaysWatchTheme : alwaysWatchTheme };
-      return angular.extend( { }, config, locals );
+      var config = angular.extend( { }, themeConfig, locals );
+
+          config.registeredStyles = [].concat(config.registeredStyles);
```
Comment (EladBezalel) [inline src/core/services/theming/theming.js]: Indention?

**Comment by jelbourn** [comment]
LGTM aside from that one indentation thing

**Comment by ThomasBurleson** [comment]
@jelbourn - fixed.

**Pull request merged**